bom_stock/models/product_product.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# © 2015 Camptocamp SA, Sodexis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, fields, api
+from openerp.addons import decimal_precision as dp
+
+
+class Product(models.Model):
+    _inherit = 'product.product'
+
+    @api.multi
+    def _bom_stock_mapping(self):
+        return {'real': 'qty_available',
+                'virtual': 'virtual_available',
+                'immediately': 'immediately_usable_qty'}
+
+    @api.one
+    def _compute_bom_stock(self):
```

Comment:
``@api.depends(
        'product_tmpl_id.bom_ids',
        'product_tmpl_id.bom_ids.bom_line_ids',
        'product_tmpl_id.bom_ids.bom_line_ids.product_id',
        'product_tmpl_id.bom_ids.bom_line_ids.product_id.qty_available',
        'product_tmpl_id.bom_ids.bom_line_ids.product_id.virtual_available',
        'product_tmpl_id.bom_ids.bom_line_ids.product_id.immediately_usable_qty',
    )
``
def _compute_bom_stock(self):

BOM manipulation triggers the functional field computation but not the qty_available, virtual_available and immediately_usable_qty changes in the BOM products.

Am I missing some feature with the @api.depends or with the relation used in depends?